File: app/src/main/java/com/sakuna63/tumbin/application/di/module/LoginPresenterModule.java
package com.sakuna63.tumbin.application.di.module;

import com.sakuna63.tumbin.application.contract.LoginContract;

import dagger.Module;
import dagger.Provides;

@Module
public class LoginPresenterModule {

    private final LoginContract.View view;

    public LoginPresenterModule(LoginContract.View view) {
        this.view = view;
    }

    @Provides
    LoginContract.View view() {
        return this.view;
    }
}

File: app/src/main/java/com/sakuna63/tumbin/application/di/module/LoginPresenterModule.kt
package com.sakuna63.tumbin.application.di.module

import com.sakuna63.tumbin.application.contract.LoginContract

import dagger.Module
import dagger.Provides

@Module
class LoginPresenterModule(private val view: LoginContract.View) {

    @Provides
    fun view(): LoginContract.View = this.view
}
